### PR TITLE
VACMS-6629: Move vamc code from va_gov_backend to va_gov_vamc.

### PR DESCRIPTION
**Pull request by ethanteague**
## Description
**Relates to #6629** 
 - moves code that is exclusive to VAMC product to `va_gov_vamc` from `va_gov_backend`
 - leverages event subscribers (instead of traditional hook system)
 - Removes code that is outdated / no longer
 - Renames function to match broader use case

## Testing done
Visual / behat

## QA steps
 - [ ] This will require a real time swarm session (~30min-ish) with @swirtSJW @netboss @ethanteague & possibly @kevwalsh

### CODE REVIEW

File: docroot/modules/custom/va_gov_backend/va_gov_backend.module [hunk 1]
```diff
@@ -1233,18 +1069,6 @@ function _va_gov_backend_dropdown_field_access(array &$form, array $targets) {
   }
 }
 
-/**
- * Implements hook_form_BASE_FORM_ID_alter().
- */
-function va_gov_backend_form_node_form_alter(array &$form, FormStateInterface &$form_state) {
-  _va_gov_backend_set_owner_section_default($form, $form_state);
```
[the remark this answers is not in the document]
Comment (swirtSJW) [reply docroot/modules/custom/va_gov_backend/va_gov_backend.module:1240]: Also I can't find where it moved to.

File: docroot/modules/custom/va_gov_backend/va_gov_backend.module [hunk 2]
```diff
@@ -741,39 +671,6 @@ function va_gov_backend_entity_view_alter(array &$build, EntityInterface $entity
     }
   }
 
-  // Check if correct content bundle and unset fields.
-  if ($entity instanceof FieldableEntityInterface && $entity->getEntityTypeId() === 'node' && $entity->bundle() === 'health_care_local_health_service') {
-    foreach (_va_gov_backend_unset_fields($entity) as $appt_intro_field) {
-      unset($build[$appt_intro_field]);
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_backend/va_gov_backend.module:747]: This went away, but I can't find where it moved.  It did not seem to get added back in.

File: docroot/modules/custom/va_gov_vamc/src/EventSubscriber/EntityEventSubscriber.php [hunk 3]
```diff
@@ -69,6 +72,42 @@ public function entityPresave(EntityPresaveEvent $event): void {
...
+   * @param array $build
+   *   The assembled entity array.
+   * @param \Drupal\Core\Entity\FieldableEntityInterface $entity
+   *   The entity.
+   */
+  public function unsetFacilityServiceFields(array &$build, FieldableEntityInterface $entity) {
+    if ($entity instanceof FieldableEntityInterface && $entity->getEntityTypeId() === 'node' && $entity->bundle() === 'health_care_local_health_service') {
+      $appt_intro_select = $entity->get('field_hservice_appt_intro_select')->value;
+      if ($appt_intro_select === 'default_intro_text') {
+        unset($build['field_hservice_appt_leadin']);
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_vamc/src/EventSubscriber/EntityEventSubscriber.php:99]: This appears here, but I can't find its removal

File: docroot/modules/custom/va_gov_vamc/src/EventSubscriber/EntityEventSubscriber.php [hunk 4]
```diff
@@ -69,6 +72,42 @@ public function entityPresave(EntityPresaveEvent $event): void {
...
+      }
+      elseif ($appt_intro_select === 'custom_intro_text') {
+        unset($build['field_hservices_lead_in_default']);
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_vamc/src/EventSubscriber/EntityEventSubscriber.php:102]: Appears, but I can not find its removal.

File: docroot/modules/custom/va_gov_vamc/src/EventSubscriber/EntityEventSubscriber.php [hunk 5]
```diff
@@ -69,6 +72,42 @@ public function entityPresave(EntityPresaveEvent $event): void {
...
+      }
+      else {
+        unset($build['field_hservice_appt_leadin']);
+        unset($build['field_hservices_lead_in_default']);
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_vamc/src/EventSubscriber/EntityEventSubscriber.php:106]: Appears, but can not find its removal.

File: docroot/modules/custom/va_gov_vamc/src/EventSubscriber/EntityEventSubscriber.php [hunk 6]
```diff
@@ -80,6 +119,20 @@ public function alterOpStatusNodeForm(FormIdAlterEvent $event): void {
...
+   * Add js script to VAMC Op status node form.
+   *
+   * Hide field_office for existing operating statuses.
+   *
+   * @param \Drupal\core_event_dispatcher\Event\Form\FormIdAlterEvent $event
+   *   The event.
+   */
+  public function alterOpStatusNodeEditForm(FormIdAlterEvent $event): void {
+    $form = &$event->getForm();
+    $form['#attached']['library'][] = 'va_gov_vamc/set_ief_administration_select';
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_vamc/src/EventSubscriber/EntityEventSubscriber.php:132]: Added but can't find removal.

**Review by swirtSJW — CHANGES_REQUESTED** [review]
This was really difficult to review with both the change of location (expected) and the refactoring (not expected)  It was really hard to account for all the removals and additions and at the same time looking at the refactoring.   I noted some things that I could not find removals and additions that match.   They may be there, but i could not find them.

**Comment by ethanteague** [comment]
I did you dirty on this one, @swirtSJW . There were a couple-few functions that were no longer being used (content model changes, etc.), that I removed. I should have broken this into several meaningful commits with meaningful messages for each item. I'll go through these and give detailed explanations.

**Comment by ethanteague** [comment]
My ambition made a mess of this - closing this pr in favor of more conservative approach: just moving from .module to .module: https://github.com/department-of-veterans-affairs/va.gov-cms/pull/6928

**Pull request closed without merging**